tests/performance-tests/src/utils/dataService.ts

```diff
@@ -0,0 +1,511 @@
+import { sleep } from 'k6';
+import http, { RefinedResponse } from 'k6/http';
+import createClient from './httpClient';
+import TestData from '../tests/testData';
+
+const applicationJsonHeaders = {
+  'Content-Type': 'application/json',
+};
+
+export interface SubjectMeta {
+  filters: {
+    [filter: string]: {
+      options: {
+        [filterGroup: string]: {
+          options: {
+            value: string;
+          }[];
+        };
+      };
+    };
+  };
+  indicators: {
+    [indicatorGroup: string]: {
+      options: {
+        value: string;
+      }[];
+    };
+  };
+  timePeriod: {
+    options: {
+      code: string;
+      year: number;
+    }[];
+  };
+  locations: {
+    [geographicLevel: string]: {
+      options: {
+        id: string;
+      }[];
+    };
+  };
+}
+
+type Topic = {
+  id: string;
+  title: string;
+  themeId: string;
+};
+
+type ThemeAndTopics = {
+  id: string;
+  title: string;
+  topics: Topic[];
+};
+
+type Release = {
+  id: string;
+  releaseName: string;
+};
+
+type PublicationAndReleases = {
+  id: string;
+  topicId: string;
+  title: string;
+  releases: Release[];
+};
+
+class DataService {
+  client;
```

Comment:
Would suggest narrowing the type of client here like so (not sure if it should be public or private in this case): 
```suggestion
  private readonly client;
```

Reply:
Yup good shout.  I've also extracted the contents of httpClient.ts into an HttpClient class, and assigned `client` here the appropriate type too.